api/tests/app/test_camera.py

```diff
@@ -0,0 +1,645 @@
+    camera_2['live_feed_enabled'] = enable
+    client.post('/cameras', json=camera_1)
+    client.post('/cameras', json=camera_2)
+    return camera_1, camera_2
+
+
+
+def get_all_cameras(config_sample_path, with_image=False):
+    config_file_json = get_config_file_json(config_sample_path, decamelize=True)
+
+    list_of_cameras = {
+        'cameras': [config_file_json[camera] for camera in config_file_json if camera.startswith('source__')]}
+
+    for camera in list_of_cameras['cameras']:
+        camera.update({'id': str(camera['id'])})
+
+        if with_image:
+            from api.routers.cameras import get_camera_default_image_string
+            image_string = get_camera_default_image_string(camera['id'])
+            # Do not forget to decode it here, to turn into string.
+            camera["image"] = image_string.decode("utf-8")
+        else:
+            camera["image"] = None
+
+    return list_of_cameras
+
+
+def get_camera_from_config_file(camera_id, config_sample_path):
+    list_of_cameras = get_all_cameras(config_sample_path)['cameras']
+    for camera in list_of_cameras:
+        if camera['id'] == camera_id:
+            return camera
+    return None
+
+
+# pytest -v api/tests/app/test_camera.py::TestClassListCameras
+class TestClassListCameras:
+    """List Cameras, GET /cameras"""
+
+    def test_get_all_cameras_no_image(self, config_rollback):
+        client, config_sample_path = config_rollback
+
+        list_of_cameras = get_all_cameras(config_sample_path)
+
+        response = client.get('/cameras')
+
+        assert response.status_code == 200
+        assert response.json() == list_of_cameras
+
+    def test_get_all_cameras_with_image(self, config_rollback):
+        client, config_sample_path = config_rollback
+
+        list_of_cameras = get_all_cameras(config_sample_path, with_image=True)
+
+        response = client.get('/cameras?options=withImage')
+
+        assert response.status_code == 200
+        assert response.json() == list_of_cameras
+
+
+# pytest -v api/tests/app/test_camera.py::TestClassCreateCamera
+class TestClassCreateCamera:
+    """Create Camera, POST /cameras"""
+
+    def test_create_one_camera_properly(self, config_rollback, camera_sample, rollback_screenshot_camera_folder):
+        client, config_sample_path = config_rollback
+
+        response = client.post('/cameras', json=camera_sample)
+
+        assert response.status_code == 201
+        for key in camera_sample:
+            if key is not "image":
+                assert response.json()[key] == camera_sample[key]
+
+    def test_try_create_camera_twice(self, config_rollback, camera_sample, rollback_screenshot_camera_folder):
+        client, config_sample_path = config_rollback
+
+        response_1 = client.post('/cameras', json=camera_sample)
+        response_2 = client.post('/cameras', json=camera_sample)
+
+        assert response_1.status_code == 201
+        assert response_2.status_code == 400
```

Comment:
i think these should validate that the nature of the error was that a camera with that id already existed

Reply:
Nice :+1:

Reply:
Reading this again I realised my message wasn't quite clear :sweat_smile:. Just in case of doubt I meant checking on the message string in the error as you did with other tests

Reply:
Perfect :)
Don t worry, it was clear. I changed it, and now it checks for the correct response.json()